api/src/test/java/org/apache/iceberg/expressions/TestExpressionUtil.java

```diff
@@ -63,6 +66,23 @@ public void testSanitizeIn() {
         ExpressionUtil.toSanitizedString(Expressions.in("test", 34, 345)));
   }
 
+  @Test
+  public void testSanitizeLongIn() {
```

Comment:
I think it would be good to have a small test around the threshold value `LONG_IN_PREDICATE_ABBREVIATION_THRESHOLD` to make sure values are abbreviated/not abbreviated.

EDIT: I accidentally noticed that something like `ExpressionUtil.toSanitizedString(Expressions.in("test", IntStream.range(0, 10).boxed().toArray()))` actually produces `test IN ((-2147483647-digit-int), (1-digit-int), ... (8 values hidden, 10 in total) ...)`. Notice the `-2147483647-digit-int` because of the `0`. FWIW, this isn't something that is being caused by this PR, so I'll take a look into this separately